### DISCUSSION TITLE
Plain text cookies

### DISCUSSION THREAD

**Discussion by runmoore** [remix-run/remix#2986] [Q&A]
Is there any advantage to using Remix's Cookie API when working with plain text cookies. My understanding is the `.parse()` function decodes the cookie which it expects has been set with the `.serialize()` function.

If the cookie is plain, nothing will be returned. Example:

```js
import { myCookie } from '~/cookies';

export async function loader({ request }) {
  const cookieHeader = request.headers.get('Cookie');
  const myCookieData = await myCookie.parse(cookieHeader);  // always null if the cookie is plain text
  ...
  return { data: myCookieData };  
}
```

Should I instead just use `cookie.parse()` from the [cookie](https://www.npmjs.com/package/cookie) lib?

**Comment by kiliman** [comment]
Remix cookies are signed and encoded. If you try to parse a text cookie, it will fail because it's not in a valid format. Definitely use another cookie lib if you need control over the value.

**Reply by runmoore** [reply]
Is there any scope for not enforcing this and allowing plain text cookies? 

The scenario I'm dealing with is a non-remix part of the site sets a plain text cookie, and I'm unable to read it using the Cookie API.
